From ed9281d6d13b6250e3f12408366fca46bc3384ce Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Thu, 26 Oct 2017 09:31:27 +0200
Subject: [PATCH] dracut.sh: after doing the lazy resolve always install
 dependencies

Exit the lazy resolving mode after doing the lazy resolve, because there
won't be any lazy resolving anymore after this step.

Fixes https://github.com/dracutdevs/dracut/issues/152
Obsoletes https://github.com/dracutdevs/dracut/pull/225
---
 dracut.sh | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/dracut.sh b/dracut.sh
index 990700af..635fdb1c 100755
--- a/dracut.sh
+++ b/dracut.sh
@@ -1544,6 +1544,10 @@ if [[ $kernel_only != yes ]]; then
         dinfo "*** Resolving executable dependencies done***"
     fi
 
+    # Now we are done with lazy resolving, always install dependencies
+    unset DRACUT_RESOLVE_LAZY
+    export DRACUT_RESOLVE_DEPS=1
+
     # libpthread workaround: pthread_cancel wants to dlopen libgcc_s.so
     for _dir in $libdirs; do
         for _f in "$_dir/libpthread.so"*; do

